libwild/src/layout.rs

```diff
@@ -733,11 +733,14 @@ trait SymbolRequestHandler<'data>: std::fmt::Display + HandlerData {
                 symbol_db.args.output_kind(),
             );
 
-            if symbol_db.args.got_plt_syms && resolution_flags.get().needs_got() {
+            if symbol_db.args.got_plt_syms
+                && resolution_flags.get().needs_got()
+                && value_flags.is_dynamic()
```

Comment:
I'd suggest removing the `is_dynamic` check. If we have PLT/GOT entries for symbols that didn't come from shared objects, then we still want symbols for them.